src/Foundation/NSAttributedStringDocumentAttributes.cs

```diff
@@ -211,15 +212,28 @@ public partial class NSAttributedStringDocumentAttributes : DictionaryContainer
 		[SupportedOSPlatform ("ios13.0")]
 		[SupportedOSPlatform ("maccatalyst")]
 		[UnsupportedOSPlatform ("tvos")]
+		[UnsupportedOSPlatform ("ios17.0")]
+		[UnsupportedOSPlatform ("maccatalyst17.0")]
+		[UnsupportedOSPlatform ("macos14.0")]
+#if __MACCATALYST__
+		[Obsolete ("Starting with maccatalyst17.0 will always return null.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif IOS
+		[Obsolete ("Starting with ios17.0 will always return null.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif MONOMAC
+		[Obsolete ("Starting with macos14.0 wll always return null.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#endif
```

Comment:
Use `[ObsoletedOSPlatform ("ios17.0", "Starting with ...")]` instead of `UnsupportedOSPlatform` + `Obsolete`.

Reply:
Looks like I am outdated